src/main/java/org/myrobotlab/jme3/Interpolator.java

```diff
@@ -1,21 +1,97 @@
 package org.myrobotlab.jme3;
 
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.myrobotlab.logging.LoggerFactory;
+import org.myrobotlab.service.JMonkeyEngine;
+import org.slf4j.Logger;
+
 public class Interpolator {
-  
-  // need to get FPS
 
-  // external thread does a request
-  public void request(String spatial, String axis, String type, Double speed) {
+  public final static Logger log = LoggerFactory.getLogger(JMonkeyEngine.class);
+
+  public class Move {
+    String name;
+    String type;
+    float newPos;
+    float speed;
+    long startTs;
+    long nextMoveTs;
+    long estimatedEndTs;
+    public Float startPos;
+    public int direction;
 
+    public String toString() {
+      return String.format("%s %s %.2f %.2f %d", name, type, newPos, speed, startTs);
+    }
   }
 
-//external thread does a request
-//processed into a data object which generates the next Msg
-  public void request(String spatial, String axis, String type, Long beginTs, Long endTs) {
+  Map<String, Move> futureMoves = new ConcurrentHashMap<String, Move>();
+  private JMonkeyEngine jme;
 
+  // need to get FPS
+
+  public Interpolator(JMonkeyEngine jMonkeyEngine) {
+    this.jme = jMonkeyEngine;
   }
-  
+
+  // external thread does a request
+  // processed into a data object which generates the next Msg
+  public void addAnimation(String name, String type, float newPos, float speed) {
+    
+    // TODO Auto-generated method stub
+    Move move = new Move();
+    // thingy we are moving
+    move.name = name;
+    move.newPos = newPos; 
+    move.startPos = jme.getAngle(name);
+    if (move.startPos == null) {
+      log.error("could not get angle for {} cannot create animation", name);
+      return;
+    }
+    move.direction = (newPos - move.startPos > 0) ? 1 : -1;
+    move.type = type;
+    move.speed = speed;
+    // FIXME - this needs to be the difference ??? - get
+                          // current pos => find distance and direction
+    // move.newPos = jme.getAngle(move.name) - newPos; // FIXME - this needs to
+    // be the difference ??? - get current pos => find distance and direction
+
+    move.startTs = System.currentTimeMillis();
+    
+    log.info("addAnimation {} {} from {} to {} @ {} degrees/sec", name, type, move.startPos, newPos, speed);
+    futureMoves.put(name, move);
+  }
+
   public Jme3Msg getNext() {
     return null;
   }
+
+  // FIXME - use ms for rate vs just seconds ?!
+  public void generateMoves() {
+    // for (Move move : futureMoves.values()) {
+    for (String name : futureMoves.keySet()) {
+      Move move = futureMoves.get(name);
+      // find the time into our move (total) vs last ..
+      long deltaTime = System.currentTimeMillis() - move.startTs; // milliseconds
+
+      // total difference in our move - degrees per second
+      float interPos = move.startPos + (move.direction * deltaTime * move.speed / 1000);
```

Comment:
It'd be really nice to have acceleration here by adding   0.5 * acceleration * deltaTime * deltaTime to this equation.

Reply:
uh .. we can't use -1 (which it currently is) so what is the average acceleration value ?

Reply:
0.0 .. no acceleration.

Reply:
I'm afraid to set it to 0.0 .. someone set it to -1 .. it looks like it might be looked at by other services (Arduino & Adafruit16)